### ISSUE TITLE
SSLStrip with bettercap 2.x against specific site not using HSTS 

### ISSUE THREAD

**Issue by ronystrom** [bettercap/bettercap#501]
This is a support issue.  I'm trying to get SSLStrip to work with Bettercap 2.x with a specific site I know should work.  I don't know if it's my usage of bettercap or an issue with the tool.  I'm running it against a site that I know doesn't properly implement HSTS, found from several blogs. The site is 'webs.com'. I know the attack is possible, because when I use my Wifi Pineapple Nano and connect a client, and browse to webs.com, it does the sslstrip attack and the user gets a browser warning. When you click through, the attack works and it shows untrusted certificate and http traffic. Now with bettercap 2.x, same site of webs.com, but sslstrip doesn't seem to work and the site is still https://.

### Environment

* Bettercap version:  2.17
* OS version:  Mac OS 10.14.3
* Command line arguments you are using.
$ sudo bettercap
» set http.proxy.sslstrip true
» set net.sniff.verbose false
» set arp.spoof.targets 192.168.7.103
» arp.spoof on
» http.proxy on
» net.sniff on
### Steps to Reproduce

1. Run bettercap commands above with victim IP address the one who is connecting to webs.com
2. Victim opens browser and types "webs.com"

**Expected behavior:  User receives a certificate warning of an untrusted certificate.  They would click through.  This is what happens when I run the same attack from a victim connected to a Wifi Pineapple Nano with PineAP module (karma).

**Actual behavior:  User doesn't receive a certificate warning.  All browsing is re-directed through 302 to https://www.webs.com

-- 

**♥ ANY INCOMPLETE REPORT WILL BE CLOSED RIGHT AWAY ♥**

**Comment by yungtravla** [comment]
> when I use my Wifi Pineapple Nano and connect a client, and browse to webs.com, it does the sslstrip attack and the user gets a browser warning. When you click through, the attack works and it shows untrusted certificate and http traffic

> Now with bettercap 2.x, same site of webs.com, but sslstrip doesn't seem to work and the site is still https://.

These are 2 different attack scenarios. If you are going to issue an invalid certificate, there is no need to sslstrip at all. You are simply forcing the victim to connect with a false certificate. SSLstrip on the other hand attempts to avoid issuing a certificate by maintaining HTTP traffic (not HTTPS).

What you want to use is the `https.proxy` in bettercap 2.x, this will start issuing your certificate which will trigger warnings in the victim's browser. Note that this is not what SSLstrip is used for.

**Comment by ronystrom** [comment]
Thanks man.  Will give this a try.  Curious, shouldn't the SSLstrip also work on this to maintain HTTP traffic between the victim browser and ettercap station?  Does this work with ettercap or is it a separate issue?

**Comment by yungtravla** [comment]
@ronystrom remember ettercap != bettercap ;)

But yes, you can enable sslstrip for the `https.proxy` by first configuring `set https.proxy.sslstrip true`.

Note that the user will first have to accept the false certificate, unless you have installed bettercap's root certificate on the victim's machine. If you have installed the root certificate, no warnings should show up and you should be able to sniff both HTTP and HTTPS traffic without any warnings on the victim's browser.

**Comment by yungtravla** [comment]
Note that these things are documented here: https://www.bettercap.org

**Comment by ronystrom** [comment]
Ha Ha, yes I know ettercap != bettercap.  Sorry!  I used to use ettercap back in the 2000s.  Bettercap is awesome.  Catching up on the tool usage for SSLStrip and other attack vectors with a modern tool.  Thanks!  I saw the other long thread from that guy who was or is having issues with SSLStrip with 2.x when he said it worked on 1.x with his tool integration (and trello.com test), so I was curious as to whether I was experiencing the same issue.  I will test it all out.  Thanks a million.  Really appreciate your replies.
